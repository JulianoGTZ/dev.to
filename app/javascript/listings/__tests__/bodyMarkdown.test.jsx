import { h } from 'preact';
import { deep } from 'preact-render-spy';
import BodyMarkdown from '../elements/bodyMarkdown';

describe('<BodyMarkdown />', () => {
  const defaultProps = {
    onChange: () => {
      return 'onChange';
    },
    default: 'defaultValue',
  };

  const renderBodyMarkdown = (props = defaultProps) =>
    deep(<BodyMarkdown {...props} />);

  it('Should match the snapshot', () => {
    const tree = renderBodyMarkdown();
    expect(tree).toMatchSnapshot();
  });
});
