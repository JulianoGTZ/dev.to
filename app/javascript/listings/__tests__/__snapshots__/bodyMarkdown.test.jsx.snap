// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BodyMarkdown /> Should match the snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<div class="field">
  <label
    class="listingform__label"
    htmlFor="body_markdown"
  >
    Body Markdown
    <textarea
      class="listingform__input listingform__bodymarkdown"
      id="body_markdown"
      name="classified_listing[body_markdown]"
      maxLength="400"
      placeholder="400 characters max, 12 line break max, no images allowed, *markdown is encouraged*"
      onInput={[Function onChange]}
    >
    </textarea>
  </label>
</div>

`;
